edk2toolext/edk2_invocable.py

```diff
@@ -151,12 +152,16 @@ def ParseCommandLineOptions(self):
         env = shell_environment.GetBuildVars()
         BuildConfig = os.path.abspath(args.build_config)
 
+        comment_re = re.compile(r"#.+$")
+
         if os.path.isfile(BuildConfig):
             with open(BuildConfig) as file:
                 for line in file:
                     stripped_line = line.strip()
-                    if not stripped_line.startswith("#"):
-                        unknown_args.append(line)
+                    stripped_line = comment_re.sub("", stripped_line)
+                    if len(stripped_line) == 0:
+                        continue
+                    unknown_args.append(stripped_line)
```

Comment:
I don't really like the BuildConfig values getting added to the unknown_args and then added to the BuildVars with the comment "From CmdLine".  

Can we update logic to correct the string so that cmd line variables and BuildConfig values are treated differently.

Reply:
Logic has been updated